internal/test/factory/validator.go

```diff
@@ -2,40 +2,33 @@ package factory
 
 import (
 	"context"
-	"fmt"
-	"math/rand"
 	"sort"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 	"github.com/tendermint/tendermint/types"
 )
 
-func RandValidator(ctx context.Context, randPower bool, minPower int64) (*types.Validator, types.PrivValidator, error) {
+func Validator(votingPower int64) (*types.Validator, types.PrivValidator, error) {
 	privVal := types.NewMockPV()
-	votePower := minPower
-	if randPower {
-		// nolint:gosec // G404: Use of weak random number generator
-		votePower += int64(rand.Uint32())
-	}
-	pubKey, err := privVal.GetPubKey(ctx)
+	pubKey, err := privVal.GetPubKey(context.Background())
```

Comment:
can we avoid this background context?

Reply:
I think we already had a context available at the call site, that we could plumb in. It's possible that might be simpler to do in a followup.

Reply:
Done :) 
Nice. That was some nice xargs + sed practice.